Build warnings

### Expected Behavior

Platform.io should build without warnings.

### Actual Behavior

Warnings are created during the build process. Some may be less important, and others may actually hint towards a problem.

The following output is generated during the build process using Platform.io:

Issue in source code:
```
Compiling .pio/build/esp32dev/src/src/battery/KIA-HYUNDAI-HYBRID-BATTERY.cpp.o
/Battery-Emulator/Software/Software.ino:150:71: warning: unsigned conversion from 'long long int' to 'uint32_t' {aka 'long unsigned int'} changes value from '1099511627775' to '4294967295' [-Woverflow]
  150 | void set(uint8_t pin, bool direction, uint32_t pwm_freq = 0xFFFFFFFFFF) {
      |                                                                       ^
```

Issue in library used:
```
Compiling .pio/build/esp32dev/src/src/lib/miwagner-ESP32-Arduino-CAN/CAN.c.o
Software/src/lib/miwagner-ESP32-Arduino-CAN/CAN.c: In function 'CAN_init':
Software/src/lib/miwagner-ESP32-Arduino-CAN/CAN.c:181:9: warning: implicit declaration of function 'gpio_matrix_out'; did you mean 'gpio_iomux_out'? [-Wimplicit-function-declaration]
  181 |         gpio_matrix_out(CAN_cfg.tx_pin_id, TWAI_TX_IDX, 0, 0);
      |         ^~~~~~~~~~~~~~~
      |         gpio_iomux_out
Software/src/lib/miwagner-ESP32-Arduino-CAN/CAN.c:182:9: warning: implicit declaration of function 'gpio_pad_select_gpio'; did you mean 'esp_rom_gpio_pad_select_gpio'? [-Wimplicit-function-declaration]
  182 |         gpio_pad_select_gpio(CAN_cfg.tx_pin_id);
      |         ^~~~~~~~~~~~~~~~~~~~
      |         esp_rom_gpio_pad_select_gpio
Software/src/lib/miwagner-ESP32-Arduino-CAN/CAN.c:186:9: warning: implicit declaration of function 'gpio_matrix_in' [-Wimplicit-function-declaration]
  186 |         gpio_matrix_in(CAN_cfg.rx_pin_id, TWAI_RX_IDX, 0);
      |         ^~~~~~~~~~~~~~
```

### Steps to Reproduce the Problem

<!-- Write some steps of how to reproduce the problem. -->

1. Build the code (a467575a) with Platform.io
2. Check the output of the build process

### Settings

<!-- Please fill in the settings used below from USER_SETTINGS.h, as it will help with diagnosis. -->

- Software version: `a467575a`

--- a/Software/src/lib/miwagner-ESP32-Arduino-CAN/CAN.c
+++ b/Software/src/lib/miwagner-ESP32-Arduino-CAN/CAN.c
@@ -266,13 +266,13 @@ int CAN_init() {
 
 bool CAN_write_frame(const CAN_frame_t *p_frame) {
 	if (sem_tx_complete == NULL) {
-		return 0;
+		return false;
 	}
 
 	// Write the frame to the controller
 	CAN_write_frame_phy(p_frame);
 
-	return xSemaphoreTake(sem_tx_complete, 20) == pdTRUE ? 1 : 0;
+	return xSemaphoreTake(sem_tx_complete, 20) == pdTRUE ? true : false;
 }
 
 int CAN_stop() {

--- a/Software/src/lib/miwagner-ESP32-Arduino-CAN/CAN.h
+++ b/Software/src/lib/miwagner-ESP32-Arduino-CAN/CAN.h
@@ -93,6 +93,10 @@ typedef struct {
 	uint8_t 			AMR3;				/**< \brief Acceptance Mask Register AMR3 */
 } CAN_filter_t;
 
+extern void gpio_matrix_in(uint32_t gpio, uint32_t signal_idx, bool inv);
+extern void gpio_matrix_out(uint32_t gpio, uint32_t signal_idx, bool out_inv, bool oen_inv);
+extern void gpio_pad_select_gpio(uint8_t gpio_num);
+
 /**
  * \brief Initialize the CAN Module
  *

--- a/Software/src/lib/miwagner-ESP32-Arduino-CAN/ESP32CAN.cpp
+++ b/Software/src/lib/miwagner-ESP32-Arduino-CAN/ESP32CAN.cpp
@@ -6,20 +6,7 @@ int ESP32CAN::CANInit() {
   return CAN_init();
 }
 bool ESP32CAN::CANWriteFrame(const CAN_frame_t* p_frame) {
-  static unsigned long start_time;
-  bool result = false;
-  if (tx_ok) {
-    result = CAN_write_frame(p_frame);
-    tx_ok = result;
-    if (!tx_ok) {
-      start_time = millis();
-    }
-  } else {
-    if ((millis() - start_time) >= 20) {
-      tx_ok = true;
-    }
-  }
-  return result;
+  return CAN_write_frame(p_frame);
 }
 
 int ESP32CAN::CANStop() {

